src/main/scala/com/mesosphere/sbt/BuildPlugin.scala

```diff
@@ -92,22 +92,25 @@ object BuildPlugin extends AutoPlugin {
     )
   }
 
-  /** These should be added to the subproject containing the main class. */
-  def itSettings(mainClassName: String): Seq[Def.Setting[_]] = {
-    oneJarSettings ++ Seq(
-      mainClass in oneJar := Some(mainClassName),
-      testOptions in IntegrationTest ++= {
-        lazy val itServer = new CosmosIntegrationTestServer(
-          (javaHome in run).value.map(_.getCanonicalPath),
-          (resourceDirectories in IntegrationTest).value,
-          oneJar.value
-        )
-
-        Seq(
-          Tests.Setup(() => itServer.setup((streams in runMain).value.log)),
-          Tests.Cleanup(() => itServer.cleanup())
-        )
-      }
+  /** This should be added to the subproject containing the main class. */
+  def allOneJarSettings(mainClassName: String): Seq[Def.Setting[_]] = {
+    oneJarSettings :+ (mainClass in oneJar := Some(mainClassName))
+  }
+
+  /** This should be appended to (testOptions in IntegrationTest) */
+  def itTestOptions(
+    javaHomeValue: Option[File],
+    classpathPrefix: Seq[File],
+    oneJarValue: File,
+    streamsValue: Keys.TaskStreams
+  ): Seq[TestOption] = {
+    val canonicalJavaHome = javaHomeValue.map(_.getCanonicalPath)
+    lazy val itServer =
+      new CosmosIntegrationTestServer(canonicalJavaHome, classpathPrefix, oneJarValue)
```

Comment:
Lazy val in a method. This is a very subtle data race. I guess it is okay since this is test code.